src/smt/optimization_solver.h

```diff
@@ -80,38 +85,163 @@ class Objective
   /** The node associated to the term that was used to construct the objective.
    * **/
   Node d_node;
-  };
-
-  /**
-   * A solver for optimization queries.
-   * 
-   * This class is responsible for responding to optmization queries. It
-   * spawns a subsolver SmtEngine that captures the parent assertions and 
-   * implements a linear optimization loop. Supports activateObjective, 
-   * checkOpt, and objectiveGetValue in that order.
-   */
-  class OptimizationSolver
-  {
-   public:
-    /** parent is the smt_solver that the user added their assertions to **/
-    OptimizationSolver(SmtEngine* parent);
-    ~OptimizationSolver();
-
-    /** Runs the optimization loop for the activated objective **/
-    OptResult checkOpt();
-    /** Activates an objective: will be optimized for **/
-    void activateObj(const Node& obj, const int& type);
-    /** Gets the value of the optimized objective after checkopt is called **/
-    Node objectiveGetValue();
-
-   private:
-    /** The parent SMT engine **/
-    SmtEngine* d_parent;
-    /** The objectives to optimize for **/
-    Objective d_activatedObjective;
-    /** A saved value of the objective from the last sat call. **/
-    Node d_savedValue;
-  };
+
+  /** Specify whether to use signed or unsigned comparison
+   * for BitVectors (only for BitVectors), this variable is defaulted to false
+   * **/
+  bool d_bvSigned;
+};
+
+/**
+ * A solver for optimization queries.
+ *
+ * This class is responsible for responding to optmization queries. It
+ * spawns a subsolver SmtEngine that captures the parent assertions and
+ * implements a linear optimization loop. Supports activateObjective,
+ * checkOpt, and objectiveGetValue in that order.
+ */
+class OptimizationSolver
+{
+ public:
+  /** parent is the smt_solver that the user added their assertions to **/
+  OptimizationSolver(SmtEngine* parent);
+  ~OptimizationSolver();
+
+  /** Runs the optimization loop for the activated objective **/
+  OptResult checkOpt();
+  /** Activates an objective: will be optimized for
+   * Parameter bvSigned specifies whether we should use signed/unsigned
+   * comparison for BitVectors (only effective for BitVectors)
+   * and its default is false **/
+  void activateObj(const Node& obj,
+                   const int& type,
```

Comment:
why int and not enum?

Reply:
it's not my thing, this is what the original optimizer looks like. if you want I will fix.